### ISSUE TITLE
Multiple unreachable code warnings if compiling with /W4 in VS2022

### ISSUE THREAD

**Issue by pps83** [boostorg/beast#2815]
Basic beast `boost_1_84_0` code produces multiple unreachable code warnings if compiling with /W4 in VS2022:

```
1>------ Build started: Project: boostbeast-warn, Configuration: Debug x64 ------
1>boostbeast-warn.cpp
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
1>D:\work-pps\boost_1_84_0\boost\beast\core\impl\buffers_cat.hpp(186): warning C4702: unreachable code
```

Here's full project to repro: https://github.com/pps83/boostbeast-warn
It's blank new console app, the only thing I changed is set `/W4` for warnings

**Comment by pps83** [comment]
This one seems to be related. It was fixed, but then later ppl said it's resurfaced: https://github.com/boostorg/beast/issues/1582

**Comment by pps83** [comment]
Locally, I've made this change to get rid of the annoying warning:

```
#pragma warning( push )
#pragma warning( disable : 4702)
        template<class I>
        reference operator()(I)
        {
            return *self.it_.template get<I::value>();
        }
#pragma warning( pop )
```

**Comment by ashtum** [comment]
Thank you for reporting, this has been fixed in #2803 and will be included for boost 1.85.

**Comment by ashtum** [comment]
Thank you for reporting. This has been fixed in #2803 and will be included in boost 1.85.

**Comment by pps83** [comment]
I tried #2803 and it does fix the error. Thanks
